src/libs/OptimisticReportNames.ts

```diff
@@ -320,21 +324,18 @@ function updateOptimisticReportNamesFromUpdates(updates: OnyxUpdate[], context:
                 if (transactionUpdate.reportID) {
                     report = getReportByID(transactionUpdate.reportID, allReports);
                 } else {
-                    report = getReportByTransactionID(getTransactionIDFromKey(update.key), context);
+                    report = getReportByTransactionID(getTransactionIDFromKey(update.key), context, workingUpdates);
```

Comment:
Shouldn't we always try to get the report from the cache, either using the reportID from the update or getting the report via the transactionID extracted from the key?

Why do we used the report without considering the cache if the update has a reportID?

Reply:
Maybe it will be more clear to me once I see how we use the report and update as it relates to the affected reports.

Reply:
After looking at this more, I don't see a reason we need to handle fetching the entire report. We should just be able to use the report ID. And since that's already the key of the affected reports, it will simplify that data storage. See my [other comment here](https://github.com/Expensify/App/pull/70870/files#r2376956214).